File: sms-service/src/main/java/com/sms/service/entity/Notification.java
package com.sms.service.entity;


import lombok.Builder;
import lombok.Data;
import lombok.extern.jackson.Jacksonized;

import java.util.Date;
import java.util.UUID;

@Data
@Builder
@Jacksonized
public class Notification {
    private final Long id;
    private final UUID uuid;
    private final NotificationMethod method;
    private final NotificationPrincipal principal;
    private final String message;
    private final Date time;
}

File: notification-service/src/main/kotlin/ua/notification/service/entity/additional/notification/Notification.kt
package ua.notification.service.entity.additional.notification

import java.util.*

data class Notification(
    val id: Long,
    val uuid: UUID,
    val notificationType: NotificationType,
    val method: NotificationMethod,
    val principal: NotificationPrincipal,
    val message: String?,
    val time: Date = Date(),
)